### ISSUE TITLE
Rerolling Fire Spirits in a specific way gives you an extra ability (6 -->7)

### ISSUE THREAD

**Issue by Draze22** [arcadia-redux/custom_hero_clash_issues#889]
![image](https://user-images.githubusercontent.com/63279022/107855811-b531c200-6e78-11eb-9169-8403617d2d73.png)

**Comment by Snoresville** [comment]
fucking hell this is like the tusk snowball bug all over again except the big boss fire spirits is here

**Comment by Eschada** [comment]
I have just tried this, it is recreatable and worked two times for me.

**Comment by 56kyle** [comment]
Just adding some specifics to how to reproduce:
1. Have fire spirits
2. Have aghanim's shard (does not matter if purchased before / after getting fire spirits)
3. Use a book of retraining and select fire spirits for being replaced.
4. Select any ability or discard, it does not matter which.
5. Quickly use another book of retraining and select fire  spirits for being replaced. This step is time sensitive.
6. Select any ability or discard and you will now have an extra ability.

tldr- it's a time sensitive thing and probably caused by a timer (I despise the timers library)

**Comment by SanctusAnimus** [comment]
Should no longer be an issue, reopen if reoccurs
